src/pudl/output/ferc1.py

```diff
@@ -1191,7 +1191,22 @@ def _out_ferc1__explosion_tags(table_dimensions_ferc1) -> pd.DataFrame:
         .reset_index()
         .drop(columns=["notes"])
     )
-    return tags_all
+    # Add the correction records to the tags with the same tags as the parent
+    idx = list(NodeId._fields)
+    correction_index = (
```

Comment:
This is "all un-corrected factoids, indexed by the NodeId fields," right? And we want to generate a bunch of tags for the corrected factoids by:

1. look at all the tags we have
2. inner merge "tags we have" with "all factoids we know about", using the NodeId fields as a join key + dropping all the random `table_dimensions` columns
3. add "_correction" to each factoid name & concat it onto all of our tags - tada!

Some questions:
1. Do we expect there to be tags for factoids that aren't in the table_dimensions? If not, can we get away with taking every tag + making a `_correction` version of it?
2. Are there factoids we expect to not have a `_correction` partner?
3. If we still need to filter out the tags we have that don't correspond to something in `table_dimensions`, should we use something more like `table_dimensions[node_id_fields].join(tags_all)` to make the intent of the code clearer?

Reply:
hmm these are all excellent questions and i think your suggestion in your first question is correct. IF what i was trying to do here was actually the right thing to do be doing. I think i could replace all of this stuff in here with `corrections = tags_all.assign(xbrl_factoid=lambda x: x.xbrl_factoid + "_correction")`

BUT I think what we really need to be doing is saying for a given calculated record, do all of the child component records contain those same tags? if so give the correction record for the parent fact those tags.

I think i was starting to attempt that using the table dims but i don't think that's the right method. hm... i could probably do it with the calculation component table but it might be simpler actually using the tree/network methods.